lib/ansible/module_utils/facts/system/service_mgr.py

```diff
@@ -143,6 +143,8 @@ def collect(self, module=None, collected_facts=None):
                 service_mgr_name = 'systemd'
             elif os.path.exists('/etc/init.d/'):
                 service_mgr_name = 'sysvinit'
+            elif os.path.exists('/sbin/runit'):
```

Comment:
Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example. Also, add tests for this change.

Reply:
Thanks for taking the time to look at the pull request. I've updated it to include a changelog fragment and two unit tests. In the first test I'm testing the situation where ``/proc/1/comm`` can be read without problems and in the second test I'm testing the situation where ``/proc/1/comm`` is unreadable and the ``ps`` command returns ``COMMAND\n`` (which is the case when it is unable to retrieve information about a process).

Reply:
Thanks.

Reply:
Thanks.

Reply:
The thing is, if reading ``/proc/1/comm`` fails, ``ps`` is also very likely to fail. I do understand you concern, I am wondering though, how this is different from the service manager fallback checks right above it. It appears to be pretty much the same situation for ``openrc`` for example.

Reply:
> also that only resolves the 'general service manager' which is what this fact tries to provide.

The reason I'm mostly interested in this is, because the ``service`` module uses this fact to decide which module to invoke for managing services. In case of ``runit`` the ``service`` module has no internal support for ``runit``. If the detection fails, ``service`` tasks also fail, since the ``service`` doesn't know what to do.

Reply:
That is why we have the `use` option, even if the 'general service manager' is chosen that does not mean the specific service you are using might be using an alternate one. 

Also I recommend using the specific service manager modules vs the generic 'service', specially now that collections exist and the names wont match (no fqcn).

Reply:
Thanks for your thoughts on the usage of the ``service`` module. Perhaps the documentation should warn about the shortcomings of ``service``, since I usually find that people recommend it over using the specific service
mamanger modules.

I always found the ``service`` module pretty convenient when writing roles that are somewhat portable between different operating systems. For typical system installations I have always found it to work quite well.

I would still be interested in your opinion on my suggestions about ways to improve the detection that I mentioned above, especially if checking whether ``/sbin/init`` is a symlink and see where it points to would be an acceptable
improvement.

Reply:
Well, for most 'standard' systems 'service' is fine, it is only an issue when you deviate from rh/ubuntu standard setups, then is where it gets 'fun'. 

The symlink detection used to be part of the code until we moved to the 'better' proc inspection, but we can readd as a fallback.

Reply:
If you are okay with it, I could modify the pull request to do the symlink detection instead.

Reply:
We might want to use a list of tools, with symlink at least we don't have the 'priority debate', taking the 'basename' will probably match correctly in most cases with a translation dict for those it doesn't (no need to do now, just add comment for the future when someone reports s-6 is misidentified.